src/handlers.rs

```diff
@@ -118,6 +118,26 @@ impl Handler {
             registration_manager: None,
         })
     }
+
+    pub async fn run(
+        mut self,
+        mut connection: mpsc::Receiver<WebSocket>,
+    ) -> Result<(), ApplicationError> {
+        // TODO: Do we want to allow a new incoming connection to replace the current one?
+        while let Some(websocket) = connection.recv().await {
+            match self.start_manager(websocket).await {
+                Ok(_) => log::info!("Manager started"),
+                Err(e) => {
+                    log::error!("Error starting the manager: {}", e);
+                }
+            }
+            // TODO in fact, this point is reached after the manager finishes
```

Comment:
Need to change the debug messages after deciding on the multi-connection question...